arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/ParTraverseTest.kt

```diff
@@ -160,4 +173,25 @@ class ParTraverseTest : ArrowFxSpec(spec = {
     val l = (0 until count).parTraverseN(20) { it }
     l shouldBe (0 until count).toList()
   }
+
+  "parSequenceN throws validation error as semaphore limit is greater than Int max value" {
+    val ref = Atomic(0)
+    shouldThrowExactly<ArithmeticException> {
+      (0 until 100)
+        .map { suspend { ref.update { it + 1 } } }
+        .parSequenceN(Long.MAX_VALUE)
+
+      ref.get() shouldBe 100
+    }
+  }
+
+  "parSequenceN can traverse effect full computations" {
```

Comment:
this finally completed in 32 minutes. :D

Reply:
Oh we probably don't want to push that test to CI 😅 I just linked it because that's what I tried locally before I realized we were using `i` from `IntRange` instead of from `update`.
It passed in 2-3 minutes locally for me.

Reply:
okay. I'll revert it :)